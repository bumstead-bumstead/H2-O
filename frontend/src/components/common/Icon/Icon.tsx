import type { SVGProps } from 'react';
import { useTheme } from '@emotion/react';
import * as icons from '@/assets/icons';

const DEFAULT_SIZE = 24;

type IconType = keyof typeof icons;

interface Props extends SVGProps<SVGSVGElement> {
  iconType: IconType;
  size?: number | string;
}

function Icon({ iconType, color, size = DEFAULT_SIZE, ...restProps }: Props) {
  const theme = useTheme();
  const Icon = icons[iconType];

  return <Icon fill={color || theme.colors.gray800} width={size} height={size} {...restProps} />;
}

export default Icon;
