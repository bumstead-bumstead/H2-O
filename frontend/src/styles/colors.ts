export const COLORS = {
  // primary
  primary100: '#CCD7E5',
  primary200: '#99B0CA',
  primary300: '#6688B0',
  primary400: '#336196',
  primary500: '#00397B',
  primary600: '#002E63',
  primary700: '#00224A',
  primary800: '#001731',
  primary900: '#000B19',

  // white
  white: '#FFFFFF',

  // grayscale
  gray50: '#F7F8F9',
  gray100: '#EDEDEE',
  gray200: '#DADCDD',
  gray300: '#C8CACC',
  gray400: '#B6B8BB',
  gray500: '#A3A6AA',
  gray600: '#838588',
  gray700: '#626466',
  gray800: '#414344',
  gray900: '#212122',

  // other colors
  activeBlue: '#00AAD2',
  activeBlue2: '#00C3F0',
  skyBlue: '#A2C7E7',
  skyBlue2: '#e3eef8',
  hoverBg: '#A2C7E70D',
  skyBlueCardBg: '#A2C7E74D',
  cardBg: '#00AAD21A',
  blueBg: '#F4F5F7',
  sand: '#B67B5E',
  sand2: '#E4DCD3',
  lightSand: '#F6F3F2',
  priceSummaryHead: '#343434',
  hashTagBg: '#75757580',
  footerBg: '#FFFFFFE5',

  // dimmed background
  dimmed: '#1F1F1FB3',
} as const;

export type ColorType = typeof COLORS;
export type Colors = keyof ColorType;
