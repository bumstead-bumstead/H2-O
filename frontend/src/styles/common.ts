export const FLEX = {
  flexDefault: {
    display: 'flex',
  },
  flexCenterCol: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  flexCenterRow: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  flexBetweenCol: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  flexBetweenRow: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  flexEndRow: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  flexEndCol: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-end',
  },
} as const;

export type FlexType = typeof FLEX;
export type FlexVariant = keyof FlexType;
