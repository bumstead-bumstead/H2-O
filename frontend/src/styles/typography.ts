export const TYPOGRAPHY = {
  HeadEnBold26: {
    fontFamily: 'Hyundai Sans Head,sans-serif',
    fontSize: '26px',
    fontWeight: 700,
    lineHeight: '36px',
    letterSpacing: '-0.01em',
  },
  HeadENBold24: {
    fontFamily: 'Hyundai Sans Head,sans-serif',
    fontSize: '24px',
    fontWeight: 700,
    lineHeight: '32px',
    letterSpacing: '-0.01em',
  },
  HeadENBold22: {
    fontFamily: 'Hyundai Sans Head,sans-serif',
    fontSize: '22px',
    fontWeight: 700,
    lineHeight: '28px',
    letterSpacing: '-0.01em',
  },
  HeadENMedium20: {
    fontFamily: 'Hyundai Sans Head,sans-serif',
    fontSize: '20px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.01em',
  },
  HeadENBold10: {
    fontFamily: 'Hyundai Sans Head,sans-serif',
    fontSize: '10px',
    fontWeight: 700,
    lineHeight: '12px',
    letterSpacing: '-0.01em',
  },

  TextENMedium18: {
    fontFamily: 'Hyundai Sans Text,sans-serif',
    fontSize: '18px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.01em',
  },
  TextENMedium16: {
    fontFamily: 'Hyundai Sans Text,sans-serif',
    fontSize: '16px',
    fontWeight: 500,
    lineHeight: '24px',
    letterSpacing: '-0.01em',
  },
  TextENMedium14: {
    fontFamily: 'Hyundai Sans Text,sans-serif',
    fontSize: '14px',
    fontWeight: 500,
    lineHeight: '22px',
    letterSpacing: '-0.01em',
  },
  CaptionENMedium14: {
    fontFamily: 'Hyundai Sans Text,sans-serif',
    fontSize: '14px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.01em',
  },
  CaptionENMedium12: {
    fontFamily: 'Hyundai Sans Text,sans-serif',
    fontSize: '12px',
    fontWeight: 500,
    lineHeight: '24px',
    letterSpacing: '-0.01em',
  },

  HeadKRBold32: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '32px',
    fontWeight: 700,
    lineHeight: '36px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold26: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '26px',
    fontWeight: 700,
    lineHeight: '36px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold24: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '24px',
    fontWeight: 700,
    lineHeight: '32px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold22: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '22px',
    fontWeight: 700,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold20: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '20px',
    fontWeight: 700,
    lineHeight: '24px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold18: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '18px',
    fontWeight: 700,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold12: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '12px',
    fontWeight: 700,
    lineHeight: '18px',
  },

  TextKRBold12: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '12px',
    fontWeight: 700,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },

  HeadKRMedium26: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '26px',
    fontWeight: 500,
    lineHeight: '36px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium24: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '24px',
    fontWeight: 500,
    lineHeight: '32px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium22: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '22px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium20: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '20px',
    fontWeight: 500,
    lineHeight: '24px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium18: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '18px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium16: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '16px',
    fontWeight: 500,
    lineHeight: '24px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium14: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '14px',
    fontWeight: 500,
    lineHeight: '22px',
    letterSpacing: '-0.03em',
  },

  TextKRMedium18: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '18px',
    fontWeight: 500,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRMedium16: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '16px',
    fontWeight: 500,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRMedium14: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '14px',
    fontWeight: 500,
    lineHeight: '22px',
    letterSpacing: '-0.03em',
  },
  TextKRMedium12: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '12px',
    fontWeight: 500,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRMedium10: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '10px',
    fontWeight: 500,
    lineHeight: '16px',
    letterSpacing: '-0.03em',
  },

  HeadKRRegular26: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '26px',
    fontWeight: 400,
    lineHeight: '36px',
    letterSpacing: '-0.03em',
  },
  HeadKRRegular24: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '24px',
    fontWeight: 400,
    lineHeight: '32px',
    letterSpacing: '-0.03em',
  },
  HeadKRRegular22: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '22px',
    fontWeight: 400,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRRegular20: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '20px',
    fontWeight: 400,
    lineHeight: '24px',
    letterSpacing: '-0.03em',
  },

  TextKRRegular18: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '18px',
    fontWeight: 400,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRRegular16: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '16px',
    fontWeight: 400,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRRegular14: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '14px',
    fontWeight: 400,
    lineHeight: '22px',
    letterSpacing: '-0.03em',
  },
  TextKRRegular12: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '12px',
    fontWeight: 400,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRRegular10: {
    fontFamily: 'Hyundai Sans Text KR,sans-serif',
    fontSize: '10px',
    fontWeight: 400,
    lineHeight: '16px',
    letterSpacing: '-0.03em',
  },

  DisplayText: {
    fontFamily: 'Hyundai Sans Head KR,sans-serif',
    fontSize: '146px',
    fontWeight: 700,
    lineHeight: '1.25',
    letterSpacing: '-0.03em',
  },
} as const;

export type TypographyType = typeof TYPOGRAPHY;
export type TypographyVariant = keyof TypographyType;
