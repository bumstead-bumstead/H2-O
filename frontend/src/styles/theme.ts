import { COLORS } from './colors';
import { FLEX } from './common';
import { TYPOGRAPHY } from './typography';

export const theme = {
  colors: COLORS,
  flex: FLEX,
  typography: TYPOGRAPHY,
};
