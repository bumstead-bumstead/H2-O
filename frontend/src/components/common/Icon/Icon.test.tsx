import Icon from './Icon';
import { render, screen } from '@/tests/test-util';
import { COLORS } from '@/styles/colors';

describe('Icon 컴포넌트 테스트', () => {
  it('Icon이 정상적으로 렌더링된다.', () => {
    render(<Icon data-testid='icon' iconType='ArrowBack' />);

    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });

  it('size props로 크기를 조절할 수 있다.', () => {
    render(<Icon data-testid='icon' iconType='ArrowBack' size='16px' />);

    const icon = screen.getByTestId('icon');
    expect(icon).toHaveAttribute('width', '16px');
    expect(icon).toHaveAttribute('height', '16px');
  });

  it('color props로 색상을 조절할 수 있다.', () => {
    render(<Icon data-testid='icon' iconType='ArrowBack' color='activeBlue' />);

    expect(screen.getByTestId('icon')).toHaveAttribute('fill', COLORS.activeBlue);
  });

  describe('Icon 컴포넌트 스냅샷 테스트', () => {
    it('ArrowBack 아이콘', () => {
      render(<Icon data-testid='icon' iconType='ArrowBack' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('ArrowDown 아이콘', () => {
      render(<Icon data-testid='icon' iconType='ArrowDown' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('ArrowLeft 아이콘', () => {
      render(<Icon data-testid='icon' iconType='ArrowLeft' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('ArrowRight 아이콘', () => {
      render(<Icon data-testid='icon' iconType='ArrowRight' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('Cancel 아이콘', () => {
      render(<Icon data-testid='icon' iconType='Cancel' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('CarbonSearch 아이콘', () => {
      render(<Icon data-testid='icon' iconType='CarbonSearch' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('Check 아이콘', () => {
      render(<Icon data-testid='icon' iconType='Check' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('InfoOutline 아이콘', () => {
      render(<Icon data-testid='icon' iconType='InfoOutline' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('Link 아이콘', () => {
      render(<Icon data-testid='icon' iconType='Link' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });

    it('Plus 아이콘', () => {
      render(<Icon data-testid='icon' iconType='Plus' />);

      expect(screen.getByTestId('icon')).toMatchSnapshot();
    });
  });
});
