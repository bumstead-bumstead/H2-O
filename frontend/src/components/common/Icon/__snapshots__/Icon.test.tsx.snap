// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > ArrowBack 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    clip-path="url(#a)"
  >
    <path
      d="M17.51 3.87 15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13Z"
      fill="inherit"
    />
  </g>
  <defs>
    <clippath
      id="a"
    >
      <path
        d="M0 0h24v24H0z"
        fill="#fff"
      />
    </clippath>
  </defs>
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > ArrowDown 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M17.835 9.165a.563.563 0 0 0-.732-.055l-.063.055-4.852 4.852-4.853-4.852a.563.563 0 0 0-.732-.055l-.063.055c-.2.2-.218.512-.055.732l.055.063 5.25 5.25c.2.2.512.218.732.055l.063-.055 5.25-5.25a.562.562 0 0 0 0-.795Z"
    fill="inherit"
  />
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > ArrowLeft 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  transform="scale(-1 1)"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    clip-path="url(#a)"
  >
    <path
      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6-6-6Z"
      fill="inherit"
    />
  </g>
  <defs>
    <clippath
      id="a"
    >
      <path
        d="M0 0h24v24H0z"
        fill="#fff"
      />
    </clippath>
  </defs>
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > ArrowRight 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    clip-path="url(#a)"
  >
    <path
      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6-6-6Z"
      fill="inherit"
    />
  </g>
  <defs>
    <clippath
      id="a"
    >
      <path
        d="M0 0h24v24H0z"
        fill="#fff"
      />
    </clippath>
  </defs>
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > Cancel 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"
    fill="inherit"
  />
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > CarbonSearch 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="m21.75 20.69-5.664-5.665a8.264 8.264 0 1 0-1.06 1.061l5.663 5.664 1.061-1.06ZM3 9.75a6.75 6.75 0 1 1 6.75 6.75A6.757 6.757 0 0 1 3 9.75Z"
    fill="inherit"
  />
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > Check 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="m9.7 18.025-5.7-5.7L5.425 10.9 9.7 15.175 18.875 6 20.3 7.425l-10.6 10.6Z"
    fill="inherit"
  />
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > InfoOutline 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M11 17h2v-6h-2v6Zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2Zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8ZM11 9h2V7h-2v2Z"
    fill="inherit"
  />
  <path
    d="M11 7h2v2h-2V7Zm0 4h2v6h-2v-6Z"
    fill="inherit"
  />
  <path
    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2Zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Z"
    fill="inherit"
  />
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > Link 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    clip-path="url(#a)"
  >
    <path
      d="M17 7h-4v2h4c1.65 0 3 1.35 3 3s-1.35 3-3 3h-4v2h4c2.76 0 5-2.24 5-5s-2.24-5-5-5Zm-6 8H7c-1.65 0-3-1.35-3-3s1.35-3 3-3h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-2Zm-3-4h8v2H8v-2Zm9-4h-4v2h4c1.65 0 3 1.35 3 3s-1.35 3-3 3h-4v2h4c2.76 0 5-2.24 5-5s-2.24-5-5-5Zm-6 8H7c-1.65 0-3-1.35-3-3s1.35-3 3-3h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-2Zm-3-4h8v2H8v-2Z"
      fill="inherit"
    />
  </g>
  <defs>
    <clippath
      id="a"
    >
      <path
        d="M0 0h24v24H0z"
        fill="#fff"
      />
    </clippath>
  </defs>
</svg>
`;

exports[`Icon 컴포넌트 테스트 > Icon 컴포넌트 스냅샷 테스트 > Plus 아이콘 1`] = `
<svg
  data-testid="icon"
  fill="#414344"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2Z"
    fill="inherit"
  />
</svg>
`;
